### PR TITLE
corner ships placement validations, add some tests, repair bugs

### PR DESCRIPTION
**Pull request by M-Aarsman**
corner ships placements validations, add some more tests and repair known issues connecting with placement

### CODE REVIEW

File: client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java [hunk 1]
```diff
@@ -62,6 +76,40 @@ void invalidPlacement(ShipOrientation shipOrientation, int shipStartIndex, int m
     assertFalse(isValid);
   }
 
+  @Test
+  void shipPassingToOtherSideShouldCauseInvalidPlacement() {
+    //given
+    ShipOrientation shipOrientation = ShipOrientation.VERTICAL;
+    int shipStartingIndex = 7;
+    int mastCount = 4;
+    FieldState[] board = producedBoard(Arrays.asList(1));
```
Comment (korotkevics) [inline client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java]: this

File: client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java [hunk 2]
```diff
@@ -62,6 +76,40 @@ void invalidPlacement(ShipOrientation shipOrientation, int shipStartIndex, int m
...
+        new ShipPlacementValidator(shipOrientation, shipStartingIndex, mastCount, board);
+
+    //when
+    boolean isValid = shipPlacementValidator.isPlacementValid();
+
+    //then
+    assertTrue(isValid);
+  }
+
   private static FieldState [] producedBoard(List<Integer> occupiedPlaces) {
```
Comment (korotkevics) [inline client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java]: Past tense for a method name? :D produceBoard

Comment (korotkevics) [reply client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java]: final to occupiedPlaces :D

File: client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java (hunk 1 above)
Comment (M-Aarsman) [reply client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java]: done

File: client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java (hunk 2 above)
Comment (M-Aarsman) [reply client/src/test/java/pl/korotkevics/ships/client/validators/ShipPlacementValidatorTest.java]: done

**Comment by korotkevics** [comment]
Fine, to proceed further with merging please firstly ask @sabath91 to take care of his changes recently added to int (not on master yet, 462 additions)

**Pull request merged**